src/lib/models/types.ts

```diff
@@ -907,11 +907,39 @@ export class ReferenceType extends Type {
         return new ReferenceType(name, target, project, name);
     }
 
+    /**
+     * In certain cases, TypeScript returns the name `default` for the name of a type that is defined with
+     * the format `export default class` or `export default function`. This method checks for that case and returns the
+     * declaration of the export instead of the name `default`.
+     */
+    private static getNameForDefaultExport(
```

Comment:
I think this is the wrong spot for this check, I'd recommend instead including it in `referenceConverter.convertType`.

Doing this in the reference converter means that we'll use the name used by the exported function rather than whatever internal name is used in the module where the symbol is declared, which is what we already do when converting type nodes.

```diff
diff --git a/src/lib/converter/types.ts b/src/lib/converter/types.ts
index 1360c137..6471cb73 100644
--- a/src/lib/converter/types.ts
+++ b/src/lib/converter/types.ts
@@ -744,9 +744,14 @@ const referenceConverter: TypeConverter<
             return ref;
         }
 
+        const name = ts.isIdentifier(node.typeName)
+            ? node.typeName.text
+            : node.typeName.right.text;
+
         const ref = ReferenceType.createSymbolReference(
             context.resolveAliasedSymbol(symbol),
             context,
+            name,
         );
         if (type.flags & ts.TypeFlags.Substitution) {
             // NoInfer<T>
```